lib/internal/perf/observe.js

```diff
@@ -171,9 +171,18 @@ function maybeIncrementObserverCount(type) {
   }
 }
 
+const kSkipThrow = Symbol('kSkipThrow');
+const performanceObserverSorter = (first, second) => {
+  return first.startTime - second.startTime;
+};
+
 class PerformanceObserverEntryList {
-  constructor() {
-    throw new ERR_ILLEGAL_CONSTRUCTOR();
+  constructor(skipThrowSymbol = undefined, entries = []) {
+    if (skipThrowSymbol !== kSkipThrow) {
+      throw new ERR_ILLEGAL_CONSTRUCTOR();
+    }
+
+    this[kBuffer] = ArrayPrototypeSort(entries, performanceObserverSorter);
```

Comment:
I don't know if it's a common case or not, but it may/may not be worth guarding the sort with a conditional that checks if `entries.length > 1` before sorting.

Reply:
I added, I got 8% faster for 1 item but degraded the `else` path by ~8%, so I didn't think that was worth it since probably this queue will be called with more items than 1.